JBBCode/ElementNode.php

```diff
@@ -123,14 +141,17 @@ public function getAsText()
      */
     public function getAsBBCode()
     {
-        $str = "[".$this->tagName;
+        $str = "[".$this->originalTagName;
         if (!empty($this->attribute)) {
-            if(isset($this->attribute[$this->tagName])) {
-                $str .= "=".$this->attribute[$this->tagName];
+            if (isset($this->attribute[$this->originalTagName])) {
+                // avoid to generate something like [FOO= bar=baz]...[/FOO]
+                if (!empty($this->attribute[$this->originalTagName]) || !$this->hasChildren()) {
+                    $str .= "=".$this->attribute[$this->originalTagName];
+                }
             }
 
-            foreach($this->attribute as $key => $value){
-                if($key == $this->tagName){
+            foreach ($this->attribute as $key => $value) {
+                if ($this->normalize($key) == $this->normalize($this->originalTagName)) {
```

Comment:
I was worried about this change by myself too. I'll go for the strcasecmp approach.